### PR TITLE
RML support on linux.

### PR DESCRIPTION
**Pull request by Ingrater**
* Major pitfall: We had two classes named ezRmlUiContext. That violates C++'s "one definition rule" and causes very strange runtime behavior on Linux.

### CODE REVIEW

File: Code/Editor/EditorEngineProcessFramework/EngineProcess/Implementation/EngineProcessDocumentContext.cpp [hunk 1]
```diff
@@ -31,6 +31,7 @@ ezEngineProcessDocumentContext* ezEngineProcessDocumentContext::GetDocumentConte
 void ezEngineProcessDocumentContext::AddDocumentContext(ezUuid guid, const ezVariant& metaData, ezEngineProcessDocumentContext* pContext, ezEngineProcessCommunicationChannel* pIPC)
 {
   EZ_ASSERT_DEV(!s_DocumentContexts.Contains(guid), "Cannot add a view with an index that already exists");
+  ezLog::Info("Adding document context {} type {}", ezArgP(pContext), pContext->GetDynamicRTTI()->GetTypeName());
```
Comment (jankrassnigg) [inline Code/Editor/EditorEngineProcessFramework/EngineProcess/Implementation/EngineProcessDocumentContext.cpp]: remove again ?

Comment (Ingrater) [inline Code/Editor/EditorEngineProcessFramework/EngineProcess/Implementation/EngineProcessDocumentContext.cpp]: TODO: Remove debug print

File: Code/ThirdParty/RmlUi/Source/Core/CMakeLists.txt [hunk 2]
```diff
@@ -47,5 +48,15 @@ if(EZ_CMAKE_PLATFORM_WINDOWS)
   )
 endif()
 
+if(EZ_CMAKE_PLATFORM_LINUX)
+  set(CMAKE_FIND_DEBUG_MODE TRUE)
```
Comment (Ingrater) [inline Code/ThirdParty/RmlUi/Source/Core/CMakeLists.txt]: TODO: Remove debug CMAKE_FIND_DEBUG_MODE

File: Code/EditorPlugins/RmlUi/EnginePluginRmlUi/RmlUiAsset/RmlUiContext.cpp [hunk 3]
```diff
@@ -16,12 +16,17 @@ EZ_BEGIN_DYNAMIC_REFLECTED_TYPE(ezRmlUiContext, 1, ezRTTIDefaultAllocator<ezRmlU
...
 // clang-format on
 
-ezRmlUiContext::ezRmlUiContext()
+ezRmlUiDocumentContext::ezRmlUiDocumentContext()
   : ezEngineProcessDocumentContext(ezEngineProcessDocumentContextFlags::CreateWorld)
 {
 }
 
-void ezRmlUiContext::OnInitialize()
+ezRmlUiDocumentContext::~ezRmlUiDocumentContext()
```
Comment (jankrassnigg) [inline Code/EditorPlugins/RmlUi/EnginePluginRmlUi/RmlUiAsset/RmlUiContext.cpp]: = default

**Review by jankrassnigg — APPROVED** [review]

File: Data/Samples/RTS/Plugins.ddl [hunk 4]
```diff
@@ -13,8 +13,3 @@ Plugin
 	string %Path{"ezRmlUiPlugin"}
 	bool %LoadCopy{false}
 }
-Plugin
```
Comment (C-Core) [inline Data/Samples/RTS/Plugins.ddl]: This should not be removed I guess?

Comment (jankrassnigg) [reply Data/Samples/RTS/Plugins.ddl]: Indeed.

**Pull request merged**